File: ios/AwesomeSuperpowered.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AwesomeSuperpowered, NSObject)

RCT_EXTERN_METHOD(construct:(NSString *)recordingPath)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(add:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/AwesomeSuperpowered.swift
@objc(AwesomeSuperpowered)
class AwesomeSuperpowered: NSObject {
    
    @objc static func requiresMainQueueSetup() -> Bool {
          return false
      }
    
    @objc func construct(recordingPath: String) -> Void {
        print(recordingPath)
    }

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b*a)
    }
    
    @objc(add:withB:withResolver:withRejecter:)
    func add(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a+b)
    }
}
